src/EditorFeatures/CSharpTest/Completion/CompletionProviders/PropertySubPatternCompletionProviderTests.cs

```diff
@@ -0,0 +1,329 @@
+            await VerifyItemExistsAsync(markup, "P4");
+        }
+
+        [Fact,]
+        public async Task PropertiesInRecursivePattern_Nested_WithFields()
+        {
+            var markup =
+@"
+public class Nested
+{
+    public int F3;
+    public int F4;
+}
+class Program
+{
+    public int P1 { get; set; }
+    public Nested P2 { get; set; }
+
+    void M()
+    {
+         _ = this is Program { P2: { $$ } }
+    }
+}
+";
+            await VerifyItemExistsAsync(markup, "F3");
+            await VerifyItemExistsAsync(markup, "F4");
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_Nested_WithMissingProperty()
+        {
+            var markup =
+@"
+public class Nested
+{
+    public int P3 { get; set; }
+    public int P4 { get; set; }
+}
+class Program
+{
+    public int P1 { get; set; }
+    public Nested P2 { get; set; }
+
+    void M()
+    {
+         _ = this is Program { : { $$ } }
+    }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_NoType()
+        {
+            var markup =
+@"
+class Program
+{
+    public int P1 { get; set; }
+    public int P2 { get; set; }
+
+    void M()
+    {
+        _ = missing is { $$ }
+    }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_MissingAfterColon()
+        {
+            var markup =
+@"
+class Program
+{
+    public int P1 { get; set; }
+    public int P2 { get; set; }
+
+    void M()
+    {
+        _ = this is { P1: $$ }
+    }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_SecondProperty()
+        {
+            var markup =
+@"
+class Program
+{
+    public int P1 { get; set; }
+    public int P2 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { P2: 1, $$ }
+    }
+}
+";
+            // VerifyItemExistsAsync also tests with the item typed.
+            await VerifyItemExistsAsync(markup, "P1");
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_NoPropertyLeft()
+        {
+            var markup =
+@"
+class Program
+{
+    public int P1 { get; set; }
+    public int P2 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { P2: 1, P1: 2, $$ }
+    }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_NotForEditorUnbrowsable()
+        {
+            var markup =
+@"
+class Program
+{
+    [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+    public int P1 { get; set; }
+
+    public int P2 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { $$ }
+    }
+}
+";
+            await VerifyItemExistsAsync(markup, "P2");
```

Comment:
you're not actually testing that `P1` is absent, which seems like it was the purpose of this test